### PR TITLE
DSA code

### PR DESCRIPTION
**Pull request by chrehall68**
This PR contains every python example. They were copied over from the google docs.

### CODE REVIEW

File: dsa/chapter1/recursion.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+def function():
+
+    # recursive call
+    function()
+
+
+# recursive function
+def offset(x):
```
Comment (chrehall68) [inline dsa/chapter1/recursion.py]: I copied this from the google doc. However, this isn't a recursive function. Should we remove this?

File: dsa/chapter2/bst.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+class BinaryTree:
+    class Node:
```
Comment (Chrovo) [inline dsa/chapter2/bst.py]: I'd put the `Node` class outside of the `BinaryTree` class. Students might get lost with indentation while reading the code and mix up what code belongs to which class. It also looks better visually in my opinion.

**Review by hpanchal092 — APPROVED** [review]
LGTM 👍

**Review by shiventi — COMMENTED** [review]
LGTM!!!

**Review by shiventi — APPROVED** [review]

**Pull request merged**